sapphire/simulations/groundparticles.py

```diff
@@ -230,14 +228,364 @@ def get_particles_in_detector(self, detector):
+                                  [200, 663.57],
+                                  [300, 936.33],
+                                  [400, 1195.46],
+                                  [600, 1686.34],
+                                  [800, 2159.36],
+                                  [1000, 2624.67],
+                                  [1500, 3757.99],
+                                  [2000, 4856.73],
+                                  [3000, 6983.24],
+                                  [4000, 9049.77],
+                                  [5000, 11063.17],
+                                  [6000, 13048.02],
+                                  [8000, 16940.54]])
+
+            energy_table = l_compton[:, 0]
+            l = l_compton[:, 1]
+
+            if gamma_energy > 7999:
+                return 171791.
+            else:
+                idx = energy_table.searchsorted(gamma_energy, side='left')
+                return l[idx]
+
+        # p [eV] and E [MeV]
+        E = p / 1e6
+
+        mips = 0
+        for energy, angle in zip(E, theta):
+            costheta = cos(angle)
+
+            # Calculate interaction point in units of scinitlator depth.
+            # If depth > 1 there is no interaction.
+
+            depth_compton = \
+                random.expovariate(1 / _compton_mean_free_path(energy))
+            depth_pair = random.expovariate(1 / _pair_mean_free_path(energy))
+
+            if ((depth_pair > scintilator_depth / costheta) &
+                    (depth_compton > scintilator_depth / costheta)):
+                # no interaction
+                continue
+
+            # Interactions in scintilator
+            elif depth_compton < depth_pair:
+                # Compton scattering
+
+                # maximum energy transfer of electron to scinitlator
+                # based on remaining scinitilator depth
+                maximum_energy_deposit_in_MIPS = \
+                    ((scintilator_depth - depth_compton) / scintilator_depth) \
+                    * max_E / MIP / costheta
+
+                # kinetic energy transfered to electron by compton scattering
+                energy_deposit_in_MIPS = _compton_energy_transfer(energy) / MIP
+
+                mips += np.minimum(maximum_energy_deposit_in_MIPS,
+                                   energy_deposit_in_MIPS)
+
+            elif energy > 1.022:
+                # Pair production: Two "electrons"
+
+                # maximum energy transfer of electron to scinitlator
+                # based on remaining scinitilator depth
+                maximum_energy_deposit_in_MIPS = \
+                    ((scintilator_depth - depth_pair) / scintilator_depth) * \
+                    max_E / MIP / costheta
+
+                # 1.022 MeV used for creation of two particles
+                # all the rest is electron kinetic energy
+                energy_deposit_in_MIPS = (energy - 1.022) / MIP
+
+                mips += np.minimum(maximum_energy_deposit_in_MIPS,
```

Comment:
`np.minimum` is element-wise minimum and thus returns a vector. 

But this is supposed to be "per item", not vectorised!

mips should be a scalar, not a vector! This needs fixing.